### PR TITLE
Add a `cosign` signing test to `sigstore-scaffolding`.

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: images/sigstore-scaffolding/tests/main.tf [hunk 1]
```diff
@@ -413,6 +416,167 @@ resource "kubernetes_job_v1" "check_rekor" {
...
+            mount_path = "/workspace"
+          }
+          volume_mount {
+            name       = "tuf-config"
+            mount_path = "/home/nonroot/.sigstore"
+          }
+        }
+
+        init_container {
+          name        = "dump-shit"
```
Comment (vaikas) [inline images/sigstore-scaffolding/tests/main.tf:477]: looks about right.

Comment (mattmoor) [reply images/sigstore-scaffolding/tests/main.tf:477]: lol

File: images/sigstore-scaffolding/tests/main.tf [hunk 2]
```diff
@@ -413,6 +416,167 @@ resource "kubernetes_job_v1" "check_rekor" {
   wait_for_completion = true
 }
 
+# Based on https://github.com/sigstore/rekor/blob/main/types.md#hashed-rekord
```
Comment (mattmoor) [inline images/sigstore-scaffolding/tests/main.tf:419]: NOTE to self: this is copy/pasta, remove before checking in.

File: images/sigstore-scaffolding/tests/main.tf [hunk 3]
```diff
@@ -125,6 +128,21 @@ resource "helm_release" "scaffold" {
...
+  // cannot handle the TUF root without the files matching
+  // these names.
+  # set {
+  #   name  = "tuf.secrets.rekor.path"
+  #   value = "rekor.pub"
+  # }
+  # set {
+  #   name  = "tuf.secrets.fulcio.path"
+  #   value = "fulcio_v1.crt.pem"
+  # }
```
Comment (mattmoor) [inline images/sigstore-scaffolding/tests/main.tf:141]: @vaikas your experiment

Comment (mattmoor) [reply images/sigstore-scaffolding/tests/main.tf]: worked, removing

File: images/sigstore-scaffolding/tests/rekor-test.tf [hunk 4]
```diff
@@ -0,0 +1,62 @@
+resource "random_pet" "test-value" {}
```
Comment (mattmoor) [inline images/sigstore-scaffolding/tests/rekor-test.tf:1]: This is just a copy of what was there before, just trying to segment things somewhat.

**Review by vaikas — APPROVED** [review]

**Pull request merged**